src/app/Dashboard/components/DashboardComponent.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import {
+  HeroComponent, NavbarComponent, SidebarComponent, ButtonComponent,
+} from '../../common/components';
+import MyStatsComponent from './MyStatsComponent';
+import SocietyStatsComponent from './SocietyStatsComponent';
+
+import { actions } from '../operations';
+import { getUserInfo } from '../../utils/tokenIsValid';
+
+export class DashboardComponent extends Component {
+  state = {
+    user: {},
+  }
+
+  /**
+   * @name defaultProps
+   * @type {PropType}
+   * @property {func} fetchUserActivites
+   *
+   */
+  static defaultProps = {
+    error: {},
+    loading: false,
+    pointsEarned: 0,
+    activitiesLogged: 0,
+    fetchUserActivites: () => {},
+  };
+
+  /**
+   * @name propTypes
+   * @type {PropType}
+   * @property {func} fetchUserActivites
+   *
+   */
+  static propTypes = {
+    loading: PropTypes.bool,
+    error: PropTypes.shape({}),
+    pointsEarned: PropTypes.number,
+    activitiesLogged: PropTypes.number,
+    fetchUserActivites: PropTypes.func,
+  };
+
+  componentDidMount() {
+    const { fetchUserActivites } = this.props;
+    const userInfo = getUserInfo();
+    this.setState({ user: userInfo });
+    fetchUserActivites(userInfo.id);
+  }
+
+  render() {
+    const {
+      error, loading, pointsEarned, activitiesLogged,
+    } = this.props;
+    const { user } = this.state;
+    if (loading) return <p>Loading ...</p>;
+    if (!loading && error) return <p>The was an error while fetching your data. Please try again later.</p>;
```

Comment:
We have talked about this before. Please evaluate the right JSX before rendering the element that you would like tp render.

Reply:
Will change that in a few.